dask/dataframe/io/csv.py

```diff
@@ -786,7 +792,32 @@ def to_csv(
             [to_csv_chunk(d, f, **kwargs) for d, f in zip(dfs[1:], files[1:])]
         )
     if compute:
-        delayed(values).compute(scheduler=scheduler)
+        if compute_kwargs is None:
+            compute_kwargs = dict()
+
+        if scheduler is not None:
+            warn(
+                "passing 'scheduler' directly into `to_csv()` is deprecated and will be removed in a future version"
+                "pass the scheduler in as part of `compute_kwargs` instead."
+                f"e.g. df.to_csv(compute=True, compute_kwargs={{scheduler={scheduler}}})",
+                FutureWarning,
+            )
+
+        if (
+            scheduler is not None
+            and compute_kwargs.get("scheduler") is not None
+            and compute_kwargs.get("scheduler") != scheduler
+        ):
+            raise ValueError(
+                f"Differing values for 'scheduler' have been passed in.\n"
+                f"scheduler argument: {scheduler}\n"
+                f"via compute_kwargs: {compute_kwargs.get('scheduler')}"
+            )
+
+        if scheduler is not None and compute_kwargs.get("scheduler") is None:
+            compute_kwargs = {"scheduler": scheduler}
```

Comment:
This will discard compute kwargs other than `"scheduler"` (e.g. `"optimize_graph"`)

```suggestion
            compute_kwargs["scheduler"] = scheduler
```

Reply:
😨 Good catch!

Reply:
I'm not sure how to write a test for this; what would be some good kwargs check against?

Reply:
Additional keyword arguments get passed to the scheduler

https://github.com/dask/dask/blob/666b53affac3ba475bd94c7ff4d29752bfa133de/dask/base.py#L405-L406

So we could test this by specifying a new entry to `compute_kwargs` like `compute_kwargs = {..., "foo": "bar"}` and then inside a custom scheduler function assert that `"foo" in kwargs`

Reply:
Test added!